assets/src/components/template-inserter/index.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Dropdown, IconButton, Spinner } from '@wordpress/components';
+import { Component } from '@wordpress/element';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { createBlock, cloneBlock } from '@wordpress/blocks';
+import { compose } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import BlockPreview from './block-preview';
+import pageIcon from './icon';
+
+const storyPageBlockName = 'amp/amp-story-page';
+
+class TemplateInserter extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.onToggle = this.onToggle.bind( this );
+
+		this.state = {
+			reusableBlocks: null,
+		};
+	}
+
+	componentDidMount() {
+		this.props.fetchReusableBlocks();
```

Comment:
@swissspidy I reworked this not to use API fetch here, however, we'll need to change it again once we get to categories.

Also looked into using the local store together with API but didn't come to a reasonable solution at this moment. Maybe for now we could actually use the API fetch as it was before and improve it later.

I'm thinking of improving the style to be acceptable and then leave the rest to the next PR-s / issues, potentially reverting to using API fetch for now. Thoughts?